### PR TITLE
Issue Fix: use-focus-trap doesn't handle Radio Groups well

### LINKED ISSUE #4678: use-focus-trap doesn't handle Radio Groups well
### What package has an issue

@mantine/hooks

### Describe the bug

When you shift + tab backwards into a Radio Group, [the last Radio button in the Group is selected](https://www.w3.org/wiki/RadioButton).
But [useFocusTrap() always calculates finalTabbable as the first Radio button in the Group](https://github.com/mantinedev/mantine/blob/33f51e1a7e755ef189d3e87ce17bd9d78a8a0742/src/mantine-hooks/src/use-focus-trap/scope-tab.ts#L9C9-L9C22).
This means it fails to `preventDefault()` or manually set the focus to the last element in `tabbable`, and the user is able to escape the focus trap.

### What version of @mantine/hooks page do you have in package.json?

6.0.19

### If possible, please include a link to a codesandbox with the reproduced problem

https://codesandbox.io/s/nervous-surf-8xdhy7?file=/src/App.tsx

### Do you know how to fix the issue

Yes

### Are you willing to participate in fixing this issue and create a pull request with the fix

Yes

### Possible fix

In `scope-tab.ts` we could check if `root.activeElement` is in a RadioGroup, and if so, check if the `finalTabbable` is in the same RadioGroup. If so, set `leavingFinalTabbable = true` as if the `activeElement` was the `finalTabbable`.

This approach is specific to Radio buttons, and I appear to be using properties not available on the `HtmlElement` interface, so it is just serving as a proof of concept. I'm open to feedback.

```typescript
export function scopeTab(node: HTMLElement, event: KeyboardEvent) {
    const tabbable = findTabbableDescendants(node)
    if (!tabbable.length) {
        event.preventDefault()
        return
    }
    const finalTabbable = tabbable[event.shiftKey ? 0 : tabbable.length - 1]
    const root = node.getRootNode() as unknown as DocumentOrShadowRoot
    let leavingFinalTabbable = finalTabbable === root.activeElement || node === root.activeElement

    // Handle the case of the active element being in a RadioGroup with the finalTabbable element
    if (root.activeElement.type === "radio") {
        const activeRadioGroup = tabbable.filter(
            (element) => element.type === "radio" && element.name === root.activeElement.name
        )
        leavingFinalTabbable = activeRadioGroup.includes(finalTabbable)
    }

    if (!leavingFinalTabbable) {
        return
    }

    event.preventDefault()

    const target = tabbable[event.shiftKey ? tabbable.length - 1 : 0]

    if (target) {
        target.focus()
    }
}
```

### PR DESCRIPTION
**Pull request by Shresth72**
Fixes #4678

```js
import { findTabbableDescendants } from './tabbable';

export function scopeTab(node: HTMLElement, event: KeyboardEvent) {
  const tabbable = findTabbableDescendants(node);
  if (!tabbable.length) {
    event.preventDefault();
    return;
  }
  const finalTabbable = tabbable[event.shiftKey ? 0 : tabbable.length - 1];
  const root = node.getRootNode() as unknown as DocumentOrShadowRoot;
  let leavingFinalTabbable = finalTabbable === root.activeElement || node === root.activeElement;

  // Handle the case of the active element being in a RadioGroup with the finalTabbable element
  const activeElement = root.activeElement as Element;
  const activeElementIsRadio =
    activeElement.tagName === 'INPUT' && activeElement.getAttribute('type') === 'radio';
  if (activeElementIsRadio) {
    const activeRadioGroup = tabbable.filter(
      (element) =>
        element.getAttribute('type') === 'radio' &&
        element.getAttribute('name') === activeElement.getAttribute('name')
    );
    leavingFinalTabbable = activeRadioGroup.includes(finalTabbable);
  }

  if (!leavingFinalTabbable) {
    return;
  }

  event.preventDefault();

  const target = tabbable[event.shiftKey ? tabbable.length - 1 : 0];

  if (target) {
    target.focus();
  }
}
```

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -6048,30 +6048,10 @@ caniuse-api@^3.0.0:
     lodash.memoize "^4.1.2"
     lodash.uniq "^4.5.0"
 
-caniuse-lite@^1.0.0, caniuse-lite@^1.0.30001449:
```
Comment (rtivital) [inline yarn.lock:6051]: Please do not downgrade caniuse-lite.

Comment (Shresth72) [reply yarn.lock:6051]: This was a previous commit, it was resolved earlier

**Comment by rtivital** [comment]
Please resolve conflicts

**Comment by rtivital** [comment]
Thanks!

**Pull request merged**